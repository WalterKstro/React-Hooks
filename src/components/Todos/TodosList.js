import TodosItems from '../Todos/TodosItems'
import PropTypes from 'prop-types';

const TodosList = ({todos, handleToogle, handleDelete}) => {
    return (
        <ol className="list-group list-group-numbered">
            {
                todos.map((todo) =>(
                    <TodosItems
                        todo={todo}
                        handleToogle={handleToogle}
                        handleDelete={handleDelete}
                    />
                ))
            }
        </ol>
    )
}

TodosList.propTypes = {
    todos: PropTypes.array.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToogle: PropTypes.func.isRequired
}

export default TodosList
