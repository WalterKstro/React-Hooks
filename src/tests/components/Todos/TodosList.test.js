import { shallow } from "enzyme"
import TodosList from "../../../components/Todos/TodosList"
import state from "../useReducer/dataTodos"

describe('Testing the component <TodosList/>', () => {

    let component
    let handleToogle
    let handleDelete
    beforeEach(() => {     
        handleToogle = jest.fn()
        handleDelete = jest.fn()   
        component = shallow(<TodosList todos={state} handleDelete={handleDelete} handleToogle={handleToogle} />)
    })


    test('should show correctly the component <TodosList/>', () => {
        expect(component).toMatchSnapshot()
    })

    test('should have same number of componentes of <TodosItems/>, to the length of the todos', () => {
        const todoItem = component.find('TodosItems').length
        expect(todoItem).toBe(state.length)
    })

    test('should has the props correctly the component <TodosItems/>', () => {
        const {todo, handleDelete, handleToogle} = component.find('TodosItems').at(0).props()
        
        expect(typeof todo).toBe(typeof state[0])
        expect(typeof handleDelete).toBe(typeof handleDelete)
        expect(typeof handleToogle).toBe(typeof handleToogle)
    })
    
    
    
})
