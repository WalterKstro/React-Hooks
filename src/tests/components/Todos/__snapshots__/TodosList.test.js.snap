// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the component <TodosList/> should show correctly the component <TodosList/> 1`] = `
<ol
  className="list-group list-group-numbered"
>
  <TodosItems
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    key="56884684698465"
    todo={
      Object {
        "id": 56884684698465,
        "state": false,
        "task": "Aprender React",
      }
    }
  />
  <TodosItems
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    key="91254684698465"
    todo={
      Object {
        "id": 91254684698465,
        "state": false,
        "task": "Aprender Vuejs",
      }
    }
  />
</ol>
`;
